src/main/java/com/grasshouse/dorandoran/post/repository/PostRepositorySupport.java

```diff
@@ -15,36 +18,69 @@
 @Repository
 public class PostRepositorySupport extends QuerydslRepositorySupport {
 
-    private JPAQueryFactory jpaQueryFactory;
+    private final JPAQueryFactory jpaQueryFactory;
 
     public PostRepositorySupport(JPAQueryFactory jpaQueryFactory) {
         super(Post.class);
         this.jpaQueryFactory = jpaQueryFactory;
     }
 
-    public Post findPostContainingComments(Long postId) {
+    public Post findPostById(Long postId) {
         return jpaQueryFactory.selectFrom(post)
-            .innerJoin(post.comments)
-            .fetchJoin()
+            .distinct()
+            .leftJoin(post.likes, postLike).fetchJoin()
+            .leftJoin(post.comments, comment).fetchJoin()
             .where(post.id.eq(postId))
+            .where(postIsAlive())
+            .where(comment.isNull().or(commentIsAlive()))
             .fetchFirst();
     }
 
-    public List<Post> findPostsInBounds(Double leftBound, Double rightBound, Double lowerBound,
-        Double upperBound) {
+    public List<Post> findPostsInBounds(Double leftBound, Double rightBound, Double lowerBound, Double upperBound) {
         return jpaQueryFactory.selectFrom(post)
+            .leftJoin(post.likes, postLike).fetchJoin()
+            .leftJoin(post.comments, comment).fetchJoin()
+            .where(postIsAlive())
             .where(betweenLatitude(lowerBound, upperBound), betweenLongitude(leftBound, rightBound))
+            .where(comment.isNull().or(commentIsAlive()))
             .fetch();
     }
 
-    public List<Post> findPostWithKeywordAndDate(String keyword, LocalDateTime startDate,
-        LocalDateTime endDate) {
-        return jpaQueryFactory
-            .selectFrom(post)
+    public List<Post> findPostWithKeywordAndDate(String keyword, LocalDateTime startDate, LocalDateTime endDate) {
```

Comment:
```suggestion
    public List<Post> findPostsWithKeywordAndDate(String keyword, LocalDateTime startDate, LocalDateTime endDate) {
```

List니 Posts로 바꾸는 건 어떨까요?